src/components/Carousel.tsx

```diff
@@ -1,18 +1,154 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './Carousel.scss';
+import { debounce } from 'lodash';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[],
+};
+
+const Carousel: React.FC<Props> = ({ images }) => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [imageSize, setImageSize] = useState(130);
+  const [visibleImages, setVisibleImages] = useState(3);
+  const [step, setStep] = useState(3);
+  const [animationDuration, setAnimationDuration] = useState(1000);
+  const [inputValue, setInputValue] = useState(130);
+
+  const wrapper = useCallback(
+    debounce((value: number) => {
+      setImageSize(value);
+    }, 1000),
+    [],
+  );
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = +event.target.value;
+
+    setInputValue(value);
+    wrapper(value);
+  };
+
+  const handleNextClick = () => {
+    const imagesLeft = images.length - currentIndex - visibleImages;
+    const nextIndex = currentIndex + Math.min(step, imagesLeft);
+
+    setCurrentIndex(nextIndex);
+  };
+
+  const handlePrevClick = () => {
+    const prevIndex = currentIndex - Math.min(step, currentIndex);
+
+    if (prevIndex >= 0) {
+      setCurrentIndex(prevIndex);
+    }
+  };
+
+  const containerWidth = visibleImages * inputValue;
+
+  const translateValue = `translateX(-${currentIndex * imageSize}px)`;
+
+  const isPrevDisabled = currentIndex === 0;
+  const isNextDisabled = currentIndex + visibleImages === images.length;
+
+  // STYLES >>>
+  const containerStylesOuter = {
```

Comment:
you could type-safe css props with respective type
```suggestion
  const containerStylesOuter: React.CSSProperties = {
```